### PR TITLE
🔎 Code consistency - Add ensure compliance section

### PR DESCRIPTION
**Pull request by bradystroud**

### CODE REVIEW

**Review by adamcogan — APPROVED** [review]
Love this ⭐️

**Review by adamcogan — CHANGES_REQUESTED** [review]
Hey Brady
CC @wicksipedia @William-Liebenberg

Love this topic to make our code neater... Let's improve the rule ⭐️ 

1. It is good to use .editorconfig however you want to be doing this from the client side (as warnings), rather than commit and then magic happens via a GitHub Action. Please make this approach the bad example

2. Also this Action looks dead. It has not been touched in 4 years. Mention it is inactive

3. The linting tool used by the action is read-only. It has been archived.

4. That linter tool looks like it only checks white space changes. ESLint and .NET can use the editorconfig file and check for a lot more

-a
www.AdamCogan.com

**Review by UlyssesMaclaren — APPROVED** [review]

**Comment by bradystroud** [comment]
> 1. It is good to use .editorconfig however you want to be doing this from the client side (as warnings), rather than commit and then magic happens via a GitHub Action. Please make this approach the bad example

Not Done - I think it is still valuable to have this on PRs. Some IDEs (vim) wont format code so the PR gate ensures all code on main is formatted.
I improved the wording to make it sound like an extra step, not a replacement to having it locally.

> 2. Also this Action looks dead. It has not been touched in 4 years. Mention it is inactive
> 3. The linting tool used by the action is read-only. It has been archived.
> 4. That linter tool looks like it only checks white space changes. ESLint and .NET can use the editorconfig file and check for a lot more

I removed the link to this action

**Review by UlyssesMaclaren — APPROVED** [review]

**Review by adamcogan — APPROVED** [review]

File: rules/consistent-code-style/rule.md [hunk 1]
```diff
@@ -15,33 +15,41 @@ guid: 292e4a99-4aa0-45f4-a9de-18ba352f9a68
...
+## Creating **.editorconfig** files
 
 **In VS 2022**
 1. Open the Add New Item dialog (Ctrl+Shift+A)
-1. Search for "EditorConfig"
-1. Select a config file depending on your project
+2. Search for "EditorConfig"
+3. Select a config file depending on your project
+
 ![Figure: Creating .editorconfig in VS 2022](vs2022-add-editorconfig.png)
```
Comment (william-liebenberg) [inline rules/consistent-code-style/rule.md:44]: Hi Brady

Your instructions are a bit on the skinny side. I'd like to see more about how to configure a good coding style and preserve it via .editorConfig and then sharing it with the team for consistency.

e.g 

To add an editor config, you can use:
- VS2022 to add a new `.editorConfig` with some of the .NET defaults as you described already
- Dotnet CLI (same defaults). 

In VS2022 you can configure your code style by going to Tools | Options | Text Editor | C# | Code Style 
From here you can generate a .editorConfig as well with all the custom VS rules that aren't part of the regular .editorConfig spec and still share them with the team.

image


I'd also note that you can use `dotnet format` to apply all the whitespace, code style and code analysis changes to the project. Or you can  run them separately as:

- `dotnet format whitespace`
- `dotnet format style`
- `dotnet format analyzers`

**Pull request merged**